### PR TITLE
Activities Editor: AddLevelsTable with Real Level Data

### PR DESCRIPTION
**Pull request by dmcavoy**
Shows the real filtered search results data in the AddLevelsDialog. This is the dialog used in the Activities Editor to add levels to Activity Sections.

Follow Up Work:
- Pagination
- Add and Clone Buttons working

Screen Shot 2020-10-01 at 8 08 45 PM

## Links

- [spec](https://docs.google.com/document/d/1LVwA7VtxkkIFZSi7StwAtgHfxFUIAN1E3nbDRjEXWP4/edit#heading=h.3lytjaku1623)
- [jira](https://codedotorg.atlassian.net/browse/PLAT-290)

## Testing story

Updated AddLevelsTableTest
Created test for summarize_for_edit
Updated tests for get_filtered_levels

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/levelbuilder/shapes.jsx [hunk 1]
```diff
@@ -19,9 +19,9 @@ export const levelShape = PropTypes.shape({
   position: PropTypes.number.isRequired,
 
   // level id, or -1 if no level is selected.
-  activeId: PropTypes.number.isRequired,
+  activeId: PropTypes.number,
   // level ids for all variants of this level
-  ids: PropTypes.arrayOf(PropTypes.number).isRequired,
+  ids: PropTypes.arrayOf(PropTypes.number),
```
Comment (davidsbailey) [inline apps/src/lib/levelbuilder/shapes.jsx:24]: why are these no longer required?

File: dashboard/app/models/levels/level.rb [hunk 2]
```diff
@@ -600,6 +600,16 @@ def summarize
     }
   end
 
+  def summarize_for_edit
```
Comment (davidsbailey) [inline dashboard/app/models/levels/level.rb:603]: nice addition!

**Review by davidsbailey — APPROVED** [review]

File: apps/src/lib/levelbuilder/shapes.jsx (hunk 1 above)
Comment (dmcavoy) [reply apps/src/lib/levelbuilder/shapes.jsx:24]: I think when you removed the level information from the activities editor this started to give errors. I think we would want this back once level information is being passed in again.

Comment (dmcavoy) [reply apps/src/lib/levelbuilder/shapes.jsx:24]: Tracking: https://codedotorg.atlassian.net/browse/PLAT-376

**Pull request merged**